packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -25,22 +32,126 @@ impl PostgresDatabase {
     }
 }
 
+impl PostgresDatabase {
+    async fn insert_version_id(&self, id: &Identifier) -> Result<(), DatastoreError> {
+        sqlx::query(r#"INSERT INTO ids (version_id, base_id) VALUES ($1, $2);"#)
+            .bind(&id.version_id)
+            .bind(&id.base_id)
+            .execute(&self.pool)
+            .await
+            .report()
+            .change_context(DatastoreError)
+            .attach_printable_lazy(|| id.clone())
+            .attach_printable("Could not insert id")?;
```

Comment:
**_Tiny_** nit: I think I'd do these two attachments in the other order

Reply:
Addressed in https://github.com/hashintel/hash/pull/734/commits/811e86c8c3ecec352ec9f303d7773b124697fc9d, can you help me understand why this ordering is important?

Reply:
When you print the report, it prints out the entire stack, and it respects the order in which you built it. As such you kind of want to build the context incrementally

- Some Error Happened
- 318903180-39131-1313801
- Could not insert id

Seems a little weird because you just have a random ID, as opposed to

- Some Error Happened
- Could not insert id
- 318903180-39131-1313801

It's not super important, and the printing might even be reversed. I just think it's good practice to say the message first, and get more specific after.